robottelo/api/utils.py

```diff
@@ -22,6 +22,16 @@
 from robottelo.exceptions import ImproperlyConfigured
 
 
+def upload_manifest(organization_id, manifest):
```

Comment:
If I remember correctly, this function changed locations because you had initially removed it and I asked you to replace it because some tests were still using it. However, it looks like you are converting some of the existing tests to access it through `target_sat`, which I didn't realize was in the scope of this PR.

Was your intention to convert all existing tests that use this method to access it through `target_sat`? If so, there are some tests missing. I did a quick search with grep:
```
 $ grep "upload_manifest(" -Ir tests/foreman/*
tests/foreman/api/test_errata.py:        upload_manifest(org.id, manifest.content)
tests/foreman/api/test_repository_set.py:        upload_manifest(org.id, manifest.content)
tests/foreman/api/test_subscription.py:        upload_manifest(module_org.id, manifest.content)
tests/foreman/api/test_subscription.py:        upload_manifest(org.id, manifest.content)
tests/foreman/api/test_subscription.py:        upload_manifest(org.id, manifest.content)
tests/foreman/api/test_subscription.py:    upload_manifest(function_org.id, manifests.original_manifest().content)
tests/foreman/api/test_subscription.py:        upload_manifest(new_org.id, manifests.clone().content)
tests/foreman/api/test_subscription.py:        upload_manifest(function_org.id, manifest.content)
tests/foreman/api/test_subscription.py:        upload_manifest(orgs[0].id, manifest.content)
tests/foreman/api/test_subscription.py:            upload_manifest(orgs[1].id, manifest.content)
tests/foreman/api/test_subscription.py:        upload_manifest(function_org.id, manifest.content)
tests/foreman/api/test_subscription.py:        upload_manifest(module_org.id, manifest.content)
tests/foreman/api/test_activationkey.py:    upload_manifest(module_org.id, session_entitlement_manifest.content)
tests/foreman/api/test_organization.py:            upload_manifest(org.id, manifest.content)
tests/foreman/api/test_product.py:        upload_manifest(org.id, manifest.content)
tests/foreman/api/test_reporttemplates.py:        upload_manifest(org.id, manifest.content)
tests/foreman/api/test_repository.py:            upload_manifest(module_org.id, manifest.content)
tests/foreman/api/test_repository.py:            upload_manifest(module_org.id, manifest.content)
tests/foreman/api/test_repository.py:#             upload_manifest(module_org.id, manifest.content)
tests/foreman/cli/test_subscription.py:        upload_manifest(module_org.id, manifest.content)
tests/foreman/cli/test_subscription.py:        upload_manifest(function_org.id, cloned_manifest.content)
tests/foreman/cli/test_subscription.py:    upload_manifest(function_org.id, manifests.original_manifest().content)
tests/foreman/destructive/test_contentview.py:        upload_manifest(org.id, manifest.content)
tests/foreman/destructive/test_repository.py:        upload_manifest(org.id, manifest.content)
tests/foreman/endtoend/test_api_endtoend.py:                upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_subscription.py:        upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_subscription.py:            upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(module_org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(org.id, manifest.content)
tests/foreman/ui/test_contentview.py:        upload_manifest(module_org.id, manifest.content)
tests/foreman/ui/test_reporttemplates.py:        upload_manifest(module_org.id, manifest.content)
```
This doesn't show which test functions are affected, but it does give us a list of the modules where it's used and the number of times it's used in each module. Could you please clarify the scope of this PR with regard to converting existing tests? If you do plan to convert all of them, then we can remove this function as part of this PR.

Reply:
@synkd The scope of this PR is to introduce new manifest_upload method but I made the changes in the test modules which were impacted.

Reply:
I think we should completely get rid of this method as part of this PR and switch all the tests to use `target_sat.upload_manifest`

Reply:
I was waiting for @synkd as there were some changes dependent on this method. I tried to replace all and removed this helper method.